### DISCUSSION TITLE
Query to reset default permission on a content type on Rhythmyx 7.3.0?

### DISCUSSION THREAD

**Discussion by mariosm1** [percussion/percussioncms#598] [Q&A] [answered]
I looked at the PSX_ACLS, ACLENTRIES, ACLENTRYPERMISSIONS tables to try to reset the permissions on a content type.  The content type doesn't seem to appear in the workbench anymore, so I'm not able to edit permissions through the UI (can't use superUserRole either).  How can I enable read access on that type again using only database queries?

**Comment by natechadwick** [comment, marked as answer]
This query will return the list of ACL, ENTRY, and Permission by content type:

```
SELECT CONTENTTYPENAME, PSX_ACLS.ID,PA.ID, PA.TYPE,P.PERMISSION FROM PSX_ACLS INNER JOIN CONTENTTYPES ON PSX_ACLS.OBJECTID = CONTENTTYPEID
INNER JOIN PSX_ACLENTRIES PA on PSX_ACLS.ID = PA.ACLID
INNER JOIN PSX_ACLENTRYPERMISSIONS P on PA.ID = P.ENTRYID
WHERE PSX_ACLS.OBJECTTYPE=2
```

Content Types are object type 2. 

To completely clear the existing ACL's on a content type you could do something like this (replace ? with the content type id of the content type you want to clear the permissions for):

```
DELETE FROM PSX_ACLENTRYPERMISSIONS WHERE
        PSX_ACLENTRYPERMISSIONS.ENTRYID IN(SELECT PSX_ACLENTRIES.ID FROM PSX_ACLENTRIES
                WHERE PSX_ACLENTRIES.ACLID =
                      (SELECT PSX_ACLS.ID FROM PSX_ACLS
                      WHERE OBJECTTYPE=2 AND OBJECTID= ?))
```

```
DELETE FROM PSX_ACLENTRIES WHERE
                                 PSX_ACLENTRIES.ACLID =
                                                 (SELECT PSX_ACLS.ID FROM PSX_ACLS
                                                  WHERE OBJECTTYPE=2 AND OBJECTID= ?)
```

```
DELETE FROM PSX_ACLS
         WHERE OBJECTTYPE=2 AND OBJECTID= ?
```

**Reply by natechadwick** [reply]
@matomario8  Did this work?
